### PR TITLE
OHM-45: Converted api/channels.coffee generators to coffeescript.

### LINKED ISSUE #341: Convert api/channels.cofffee (med) to coffeescript
See epic here: https://github.com/jembi/openhim-core-js/issues/200

### PR DESCRIPTION
**Pull request by rcrichton**
Closes #341 

@devcritter could you review this to? Else feel free to hand it off if I've given you too much CR :)

### CODE REVIEW

File: src/api/channels.coffee [hunk 1]
```diff
@@ -198,42 +160,33 @@ processPostDeleteTriggers = (channel) ->
...
+    return
+
+  # Get the values to use
+  id = unescape channelId
+
+  try
+    numExistingTransactions = yield Transaction.count({ channelID: id }).exec()
+
+    # Try to get the channel (Call the function that emits a promise and Koa will wait for the function to complete)
+    channel
```
Comment (devcritter) [inline src/api/channels.coffee]: This can be removed. It'll just translate to a one line `channel;` in js, not a var declaration.

**Comment by devcritter** [comment]
Naah, it's alright.

Looks good - just a rogue channel statement

File: src/api/channels.coffee (hunk 1 above)
Comment (rcrichton) [inline src/api/channels.coffee]: Ah, oops, must have missed that. Thanks.

**Comment by rcrichton** [comment]
Ok @devcritter fixed :)

**Comment by devcritter** [comment]
:+1:

**Pull request merged**